lms/djangoapps/discussion/rest_api/api.py

```diff
@@ -1250,3 +1257,72 @@ def delete_comment(request, comment_id):
         comment_deleted.send(sender=None, user=request.user, post=cc_comment)
     else:
         raise PermissionDenied
+
+
+@function_trace("_get_course_discussion_user_stats")
+def _get_course_discussion_user_stats(
+    course_key: CourseKey,
+    order_by: UserOrdering = UserOrdering.BY_FLAGS,
+) -> List[Dict[str, Union[str, int]]]:
+    """
+    Get course user discussion stats.
+
+    Args:
+        course_key (CourseKey): course key for course to fetch discussion stats for.
+        order_by (UserOrdering): the ordering to use for sorting data
+
+    Returns:
+        Sorted user stat data for the specified course
+    """
+    course_stats = get_course_user_stats(course_key)
+
+    def key_func(user):
+        key = (user["threads"], user["responses"], user["replies"])
+        if order_by == UserOrdering.BY_FLAGS:
+            key = (user["active_flags"], user["inactive_flags"]) + key
+        return key
+
+    return sorted(course_stats.values(), key=key_func, reverse=True)
+
+
+@function_trace("get_course_discussion_user_stats")
+def get_course_discussion_user_stats(
+    request,
+    course_key_str: str,
+    page: int,
+    page_size: int,
+    order_by: UserOrdering = None,
+) -> Dict:
+    """
+    Get paginated course discussion stats for users in the course.
+
+    Args:
+        request (Request): DRF request
+        course_key_str (str): course key string
+        page (int): Page number to fetch
+        page_size (int): Number of items in each page
+        order_by (UserOrdering): The ordering to use for the user stats
+
+    Returns:
+        Paginated data of a user's discussion stats sorted based on the specified ordering.
+
+    """
+    course_key = CourseKey.from_string(course_key_str)
+    if has_discussion_privileges(user=request.user, course_id=course_key):
+        order_by = order_by or UserOrdering.BY_FLAGS
+    else:
+        order_by = order_by or UserOrdering.BY_ACTIVITY
+        if order_by != UserOrdering.BY_ACTIVITY:
```

Comment:
rather than raise an error here, can we simply update the `order_by` value here as this is the only allowed value when a user doesn't have permission?

Reply:
I think that is a bit risky, since it essentially means we allow arbitrary values for `order_by`, and doesn't convey to the user that their request isn't being honoured.

Reply:
I don't think its risky as its just ordering the available stats for user but I'll go with what you think the best. 

secondly, I'll also suggest we catch & update this value early in the view and `get_course_discussion_user_stats` shouldn't be responsible for deciding what is the valid order_by value is for this user.

Reply:
The advantage of adding this logic in this function is that the view is then dumb and doesn't really need to be tested. That function contains all the logic we need so can be tested easily.